src/Polly.Extensions/DependencyInjection/PollyServiceCollectionExtensions.cs

```diff
@@ -238,7 +238,7 @@ private static void AddResiliencePipelineBuilder(this IServiceCollection service
             .AddOptions<TelemetryOptions>()
             .Configure<IServiceProvider>((options, serviceProvider) =>
             {
-                options.LoggerFactory = serviceProvider.GetService<ILoggerFactory>() ?? NullLoggerFactory.Instance;
+                options.LoggerFactory ??= serviceProvider.GetService<ILoggerFactory>() ?? NullLoggerFactory.Instance;
```

Comment:
This change will actually never use the factory from DI because it has a default value already assigned.

I'll take another look and try to replicate your problem during next week.